src/Application.vala

```diff
@@ -98,6 +98,17 @@ public class AppCenter.App : Gtk.Application {
             return;
         }
 
+        if (file.has_uri_scheme ("type")) {
+            string? mimetype = mimetype_from_file (file);
+            if (mimetype != null) {
+                main_window.search (mimetype);
```

Comment:
This performs a general search for the mimetype string, not specifically for an app that takes that mimetype as a parameter.  It may therefore throw up irrelevant results.